### PR TITLE
Get SslHandler by name for ALPN in HttpClientChannelInitializer

### PR DESCRIPTION
**Pull request by j-bahr**
Switch to accessing the SslHandler by name in client ALPN. This should always return the SslHandler for remote endpoint. If we don't do this, the pipeline will always return the first SslHandler in the pipeline. When talking through a TLS enabled forward proxy there may be more than one SslHandler present in the pipeline. This change ensures that we always get the SslHandler for the remote HTTP endpoint and not an intermediary, ensuring we set the correct negotiated protocol.

Fixes reactor/reactor-netty#2480

### CODE REVIEW

File: reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientConfig.java [hunk 1]
```diff
@@ -867,27 +867,30 @@ static final class H2OrHttp11Codec extends ChannelInboundHandlerAdapter {
...
+				}
+				else {
+					throw new IllegalStateException("unknown protocol: " + protocol);
+				}
 
-			ctx.channel().pipeline().remove(this);
+				ctx.fireChannelActive();
+
+				ctx.channel().pipeline().remove(this);
+			} else {
```
Comment (violetagg) [inline reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientConfig.java:891]: Make checkstyle happy
`[checkstyle] [ERROR] /.../reactor-netty/reactor-netty/reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientConfig.java:891:25: '}' at column 4 should be alone on a line. [RightCurly]`

```suggestion
			}
			else {
```

**Review by violetagg — CHANGES_REQUESTED** [review]

**Comment by violetagg** [comment]
@reactor/netty-team PTAL

**Comment by violetagg** [comment]
@reactor/netty-team The PR is targeting `main` branch. I will manually apply it to `1.0.x` branch.

**Review by pderop — APPROVED** [review]
it looks good to me, thanks

**Comment by violetagg** [comment]
@j-bahr Thanks for the PR.
I've merged the fix to `1.0.x` branch (f8f9a3d) and then forward merged to `main` branch (2662ecc) and `netty5` branch (a51c5b7)

**Pull request closed without merging**